cohortextractor2/backends/tpp.py

```diff
@@ -59,14 +59,38 @@ class TPPBackend(BaseBackend):
     patient_join_column = "Patient_ID"
 
     patients = MappedTable(
-        implements=tables.Patients,
         source="Patient",
         columns=dict(
             sex=Column("varchar", source="Sex"),
             date_of_birth=Column("date", source="DateOfBirth"),
         ),
     )
 
+    patient_demographics = QueryTable(
+        implements=tables.PatientDemographics,
+        columns=dict(
+            sex=Column("varchar"),
+            date_of_birth=Column("date"),
+            date_of_death=Column("date"),
+        ),
+        query="""
+            SELECT  Patient_ID as patient_id,
+                DateOfBirth as date_of_birth,
+                CASE
+                    WHEN DateOfDeath = '9999-12-31' THEN NULL
```

Comment:
This will need to be addressed again later; we may want to keep ceiling dates (in which case we need to not replace the day, and validation needs to deal with them), and/or we may need to replace nulls with ceiling dates